code/game/turfs/simulated/walls.dm

```diff
@@ -168,63 +170,74 @@
 	var/turf/T = user.loc	//get user's location for delay checks
 
 	//the istype cascade has been spread among various procs for easy overriding
-	if(try_wallmount(W,user,T) || try_decon(W,user,T) || try_destroy(W,user,T))
+	if(try_clean(W, user, T) || try_wallmount(W, user, T) || try_decon(W, user, T) || try_destroy(W, user, T))
 		return
 
+/turf/closed/wall/proc/try_clean(obj/item/W, mob/user, turf/T)
+	if((user.a_intent != INTENT_HELP) || !damage_decals.len || !istype(W, /obj/item/weldingtool))
+		return FALSE
+	var/obj/item/weldingtool/WT = W
+	if(WT.remove_fuel(0, user))
+		to_chat(user, "<span class='notice'>You begin fixing dents on the wall...</span>")
+		playsound(src, W.usesound, 100, 1)
+		if(do_after(user, slicing_duration * W.toolspeed * 0.5, target = src))
+			if((iswallturf(src) || !user || !WT || !WT.isOn() || T) && (user.loc == T) && (user.get_active_held_item() == WT) && damage_decals.len )
```

Comment:
you should check !QDELETED(WT) and !QDELETED(T)

Reply:
as opposed to !WT and !T